### PR TITLE
Improvements to type stability

### PR DESCRIPTION
**Pull request by odow**
Roughly 40% faster with these changes.

@EyobZewdie this should help speed your model up

### CODE REVIEW

File: src/Lagrangian/levelmethod.jl [hunk 1]
```diff
@@ -85,7 +85,7 @@ function lagrangian_method!{S,T}(lp::LinearProgramData{LevelMethod{S,T}}, m::JuM
     while iteration < levelmethod.maxit
         iteration += 1
         # Evaluate the real function and a subgradient
-        # m.internalModelLoaded = false
+        m.internalModelLoaded = false # need to do this smarter...
```
Comment (odow) [inline src/Lagrangian/levelmethod.jl]: why is this necessary?

File: src/Lagrangian/levelmethod.jl [hunk 2]
```diff
@@ -125,12 +125,12 @@ function lagrangian_method!{S,T}(lp::LinearProgramData{LevelMethod{S,T}}, m::JuM
         end
         # Form a level
         if dualsense == :Min
-            level = f_approx + (gap * lp.method.level + tol.val/10)
-
+            level = f_approx + (gap * lp.method.level + tol.val/10.0)
+            setupperbound(θ, level)
```
Comment (odow) [inline src/Lagrangian/levelmethod.jl]: oops. I didn't look carefully enough

File: src/Lagrangian/levelmethod.jl (hunk 1 above)
Comment (lkapelevich) [reply src/Lagrangian/levelmethod.jl]: I forgot about this... there is one model and the objective changes between being linear and quadratic. If we don't want to use the quadratic solver for the linear part of this, we can't keep the quadratic terms.
Normally we would go into 
https://github.com/JuliaOpt/JuMP.jl/blob/master/src/quadexpr.jl#L69

Comment (odow) [reply src/Lagrangian/levelmethod.jl]: Which model?
